libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/QueriesProvider.kt

```diff
@@ -0,0 +1,14 @@
+package com.mapbox.navigation.base.internal.utils
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+
+/**
+ * Queries params, which add data to [DirectionsResponse] and the SDK cannot faked it
+ * (See [https://docs.mapbox.com/api/navigation/directions](https://docs.mapbox.com/api/navigation/directions)):
+ * - `engine=electric` - electronic vehicle routing. [https://docs.mapbox.com/api/navigation/directions/#electric-vehicle-routing](https://docs.mapbox.com/api/navigation/directions/#electric-vehicle-routing)
+ */
+internal object QueriesProvider {
+    internal fun provideQueries(): Map<String, String> = mapOf(
```

Comment:
Why create a new map every time? Can't it be just val?

Reply:
not needed. The map is consumed by `DirectionsRouteMissingConditionsCheck` and stored there. I added the provider to have the option to mock it for tests